components/AddressInput.tsx

```diff
@@ -75,16 +73,13 @@ const AddressInput = ({
     userIsSender ? 'border-bt-300' : 'border-gray-300'
   )
 
-  const onAddressChange = useCallback(
-    async (event: React.SyntheticEvent) => {
-      const data = event.target as typeof event.target & {
-        value: string
-      }
-      setValue(data.value.trim())
-      onInputChange && onInputChange(event)
-    },
-    [onInputChange]
-  )
+  const onAddressChange = async (event: React.SyntheticEvent) => {
```

Comment:
So redefine does not always trigger a render, it would only re-render if it was into a useEffect dependency array or passed as a prop. useCallback is pretty useless when it comes to instances where we just have a function that we call without it being required in the dependency array. 

An interesting read around it - https://kentcdodds.com/blog/usememo-and-usecallback

Also, I am still seeing if it all actually benefits, as of now the number of re-renders have dropped significantly

Reply:
It's fair that `useCallback` does use more CPU and is harder to garbage collect than without. But nothing in that article indicates that removing `useCallback` can lead to more re-renders. It shouldn't.

I'm skeptical the re-render reductions you are seeing are related to these specific `useCallback`s, given how many other changes are in this PR.

The `input` that takes the reference to `onAddressChange` will get a new pointer for `onAddressChange` on every render of the component because we are defining a new function each time. If it had children, those would all have to re-render as well (it doesn't in this case, because it's a HTML element and not a user-defined react component). I'd worry more about the issue on components higher up in the tree. For example `Layout.tsx`, where the additional re-renders will be significantly more expensive than the saved CPU cycles of wrapping the function and defining the array.

Reply:
> it would only re-render if it was into a useEffect dependency array or passed as a prop. 

That's all true. But a lot of the usages that were removed in this PR are passed as props

Reply:
I agree, but I see using callback did not reduce the number of re-renders, I am still investigating, to see where it made sense to add it and will add them back. Just as an approach removed all to see how it goes